airflow/operators/datetime.py

```diff
@@ -64,10 +66,16 @@ def __init__(
         self.target_upper = target_upper
         self.follow_task_ids_if_true = follow_task_ids_if_true
         self.follow_task_ids_if_false = follow_task_ids_if_false
-        self.use_task_execution_date = use_task_execution_date
+        self.use_task_logical_date = use_task_logical_date
+        if use_task_execution_date:
```

Comment:
Would it be easier to use `exactly_one` here?

Reply:
I think `exactly_one` fits more to mutually exclusive + both are valid options.
Here while we want  mutually exclusive only 1 option is valid since the other is deprecated.
Using `exactly_one` won't save us the if condition as we still need to raise deprecation warning.

WDYT?